apps/authentication-service/src/shared/tokenUtils.test.ts

```diff
@@ -0,0 +1,144 @@
+import { getTestToken } from "@/test/testData/users";
+import fs from "fs";
+
+import {
+  getCachedToken,
+  getIsTokenExpired,
+  getLocalToken,
+  logToken,
+  setCachedToken,
+  setLocalToken,
+} from "@/shared/tokenUtils";
+
+let tokenBackup: string;
+const validTestToken: string = getTestToken();
+const expiredTestToken: string = getTestToken(true);
+
+describe("tokenUtils", () => {
+  beforeAll(() => {
+    tokenBackup = getCachedToken() || "";
+  });
+
+  afterAll(() => {
+    if (tokenBackup) {
+      setCachedToken(tokenBackup);
+    }
+    jest.clearAllMocks();
+  });
+
+  describe("setCachedToken", () => {
+    it("sets token", () => {
+      jest.spyOn(fs, "existsSync").mockReturnValue(true);
+      jest.spyOn(fs, "writeFileSync").mockImplementation(() => {});
+      jest.spyOn(fs, "readFileSync").mockReturnValue(validTestToken);
+      setCachedToken(validTestToken);
+      expect(getCachedToken()).toBe(validTestToken);
```

Comment:
This line doesn't give us any confidence. We mocked writeFileSync, so it doesn't actually write to the file, so calling getCachedToken doesn't tell us that anything happened. 

The only way we can get any confidence with this particular test is to make sure writeFileSync was called with the correct parameters.

Reply:
When you get rid of this line, then you can get rid of the mocks for existsSync and readFileSync

Reply:
Here's my new attempt:

```
  describe("setCachedToken", () => {
    it("sets token to the cache file", () => {
      const testToken = "setCachedTokenTestToken";
      jest.spyOn(fs, "writeFileSync");
      setCachedToken(testToken);
      expect(fs.writeFileSync).toHaveBeenCalledWith(tokenFile, testToken);
    });
  });
```

Reply:
Looks good